tests/test_version_range.py

```diff
@@ -209,9 +211,28 @@ def test_PypiVersionRange_raises_ivr_for_invalid_ranges(self):
         except InvalidVersionRange as ivre:
             assert str(ivre).startswith("Unsupported character")
 
+    def test_NpmVersionRange_from_native_with_compatible_with_version_operator(self):
+        npm_range = "^1.2.9"
+        expected = (
+            VersionConstraint(comparator=">=", version=SemverVersion(string="1.2.9")),
+            VersionConstraint(comparator="<", version=SemverVersion(string="2.0.0")),
+        )
+        version_range = NpmVersionRange.from_native(npm_range)
+        assert version_range.constraints == expected
+
+    def test_NpmVersionRange_from_native_with_approximately_equal_to_operator(self):
+        npm_range = "~3.8.2"
+        expected = (
+            VersionConstraint(comparator=">=", version=SemverVersion(string="3.8.2")),
+            VersionConstraint(comparator="<", version=SemverVersion(string="3.9.0")),
+        )
+        version_range = NpmVersionRange.from_native(npm_range)
+        assert version_range.constraints == expected
+
 
 VERSION_RANGE_TESTS_BY_SCHEME = {
     "nginx": ["0.8.40+", "0.7.52-0.8.39", "0.9.10", "1.5.0+, 1.4.1+"],
+    "npm": ["^1.2.9", "~3.8.2"],
```

Comment:
Can you please add more ranges for npm ? It would really help make other tests better.